core/src/test/scala/org/apache/spark/scheduler/TaskContextSuite.scala

```diff
@@ -228,6 +228,31 @@ class TaskContextSuite extends SparkFunSuite with BeforeAndAfter with LocalSpark
     assert(res === Array("testPropValue,testPropValue"))
   }
 
+  test("immediately call a completion listener if the context is completed") {
+    var invocations = 0
+    val context = TaskContext.empty()
+    context.markTaskCompleted()
+    context.addTaskCompletionListener(_ => invocations += 1)
+    assert(invocations == 1)
+    context.markTaskCompleted()
+    assert(invocations == 2)
```

Comment:
can we call `context.markTaskCompleted()` once again and assert `invocations == 2` to have a test for idempotency?